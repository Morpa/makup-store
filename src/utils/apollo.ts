import { ApolloClient, HttpLink, NormalizedCacheObject } from '@apollo/client'
import { useMemo } from 'react'

import apolloCache from './apolloCache'

let apolloClient: ApolloClient<NormalizedCacheObject>

function createApolloClient() {
  const httpLink = new HttpLink({
    uri: `${process.env.NEXT_PUBLIC_API_URL}/graphql/`
  })

  return new ApolloClient({
    ssrMode: typeof window === 'undefined',
    link: httpLink,
    cache: apolloCache
  })
}

export function initializeApollo(initialState = {}) {
  const apolloClientGlobal = apolloClient ?? createApolloClient()

  if (initialState) {
    apolloClientGlobal.cache.restore(initialState)
  }

  if (typeof window === 'undefined') return apolloClientGlobal

  apolloClient = apolloClient ?? apolloClientGlobal

  return apolloClient
}

export function useApollo(initialState = {}) {
  const store = useMemo(() => initializeApollo(initialState), [initialState])
  return store
}
