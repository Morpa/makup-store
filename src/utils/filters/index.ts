import { ParsedUrlQueryInput } from 'querystring'

import { ItemProps } from 'components/ExploreSidebar'

type ParseArgs = {
  queryString: ParsedUrlQueryInput
  filterItems: Pick<ItemProps, 'type' | 'name'>[]
}

export const parseQueryStringToFilter = ({
  queryString,
  filterItems
}: ParseArgs) => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const obj: any = {}

  Object.keys(queryString).forEach((key) => {
    const item = filterItems?.find((item) => item.name === key)
    const isCheckbox = item?.type === 'checkbox'
    const isArray = Array.isArray(queryString[key])

    obj[key] = !isArray && isCheckbox ? [queryString[key]] : queryString[key]
  })

  return obj
}
