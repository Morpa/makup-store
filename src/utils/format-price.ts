export default function formatPrice(price: number | bigint): string {
  return new Intl.NumberFormat('pt-PT', {
    style: 'currency',
    currency: 'EUR'
  }).format(price)
}
