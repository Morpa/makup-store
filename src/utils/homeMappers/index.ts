import { ProductApiResponse } from 'services/api'

export const productsMapper = (products: ProductApiResponse[]) => {
  return products
    ? products.map((item) => {
        return {
          id: item.id,
          title: item.name,
          brand: item.brand?.toLocaleUpperCase(),
          img: `https:${item.api_featured_image}`,
          price:
            Number(item.price) > 0
              ? item.price
              : Math.floor(Math.random() * 50),
          rating: Number(item?.rating) > 0 ? Number(item.rating) : Number(0)
        }
      })
    : {}
}

export const bannersMapper = (products: ProductApiResponse[]) => {
  return products
    ? products.map((item) => {
        return {
          img: `https:${item.api_featured_image}`,
          title: item.name,
          buttonLabel: 'Buy now',
          buttonLink: `/product/${item.id}`
        }
      })
    : {}
}

export const highlightMapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  products: any,
  title: string,
  subtitle: string,
  alignment: string
) => {
  return products
    ? products.map((item: ProductApiResponse) => {
        return {
          title,
          subtitle,
          backgroundImage: `https:${item.api_featured_image}`,
          buttonLabel: 'Buy now',
          buttonLink: `/product/${item.id}`,
          alignment
        }
      })
    : {}
}
