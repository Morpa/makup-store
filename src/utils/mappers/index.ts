import { QueryProducts_products_product } from 'graphql/generated/QueryProducts'

export const productsMapper = (products: QueryProducts_products_product[]) => {
  return products
    ? products.map((item) => {
        return {
          id: item.id,
          title: item.name,
          brand: (item.brand?.length && item.brand?.toLocaleUpperCase()) || '',
          img: `https:${item.api_featured_image}`,
          price:
            Number(item.price) > 0
              ? item.price
              : Math.floor(Math.random() * 50),
          rating: Number(item?.rating) > 0 ? Number(item.rating) : Number(0)
        }
      })
    : {}
}

export const bannersMapper = (products: QueryProducts_products_product[]) => {
  return products
    ? products.map((item) => {
        return {
          img: `https:${item.api_featured_image}`,
          title: item.name,
          buttonLabel: 'Buy now',
          buttonLink: `/product/${item.id}`
        }
      })
    : {}
}

export const highlightMapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  products: any,
  title: string,
  subtitle: string,
  alignment: string
) => {
  return products
    ? products.map((item: QueryProducts_products_product) => {
        return {
          title,
          subtitle,
          backgroundImage: `https:${item.api_featured_image}`,
          buttonLabel: 'Buy now',
          buttonLink: `/product/${item.id}`,
          alignment
        }
      })
    : {}
}
