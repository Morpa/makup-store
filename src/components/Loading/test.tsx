import { render, screen } from 'utils/test-utils'

import { Loading } from '.'

describe('<Loading />', () => {
  it('should render the heading', () => {
    const { container } = render(<Loading />)

    expect(
      screen.getByRole('img', {
        name: /Loading more products/i
      })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
