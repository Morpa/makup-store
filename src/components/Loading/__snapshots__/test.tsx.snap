// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render the heading 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 3.2rem;
}

.c1 {
  color: #FAFAFA;
  font-size: 1.8rem;
  font-weight: 300;
  margin-bottom: 3.2rem;
}

@media (max-width:768px) {
  .c0 {
    min-width: 22rem;
  }
}

<div
  class="c0"
>
  <img
    alt="Loading more products"
    height="60"
    role="img"
    src="/img/loading.gif"
    width="60"
  />
  <p
    class="c1"
  >
    Loading...
  </p>
</div>
`;
